lms/djangoapps/teams/serializers.py

```diff
@@ -160,3 +162,6 @@ def __init__(self, *args, **kwargs):
         topics_to_team_count = {d['topic_id']: d['team_count'] for d in teams_per_topic}
         for topic in self.data['results']:
             topic['team_count'] = topics_to_team_count.get(topic['id'], 0)
+
+        if 'sort_order' in self.context and self.context['sort_order'] == 'team_count':
```

Comment:
I think this approach may be reversed. It looks to me like the alphabetical sort is currently the primary, while the secondary sort is `team_count`: first we sort all the topics on the course module alphabetically in `get_alphabetical_topics`, we then cut off a page of results, and hand it to the numerical `team_count` sort which only sorts within the page.

 This will sort the current page by `team_count`, but if the current page itself only includes topics with low `team_count`s, then we're not really sorting by team count.